### PR TITLE
Initial setup for wjb instances with Jupiter Brain on MacStadium

### PR DESCRIPTION
**Pull request by henrikhodne**
**WIP**: I (@henrikhodne) am applying frequent changes locally as I'm working on this, so please ping me if you want to work on it locally before this is merged.

## TODO

- [x] Make the IPs for `wjb` slightly more dynamic (since we'll spin up more than one of them). Probably using the IP manipulation functions in Terraform with a bit of math
- [x] Add the index to `wjb`'s name, so it'll be `wjb-1` and `wjb-2`
- [x] Add a module to install/configure workers (should be pretty similar to Jupiter Brain module)
- [x] Add a module to install/configure HAProxy
- [x] Change the `wjb` resource to add a provisioner to install the Threat Stack client
- [x] Log output from workers to syslog
- [x] Add collectd-vsphere
- [x] Add ability to add custom jupiter-brain+worker pairs
- [x] Add vsphere-janitor

### CODE REVIEW

**Comment by solarce** [comment]
Also,

- [ ] add `vsphere-janitor`

File: modules/haproxy/haproxy.tf [hunk 1]
```diff
@@ -0,0 +1,55 @@
+variable "ssh_host" {}
+variable "ssh_user" {}
+variable "config" { type = "list" }
```
Comment (igorwwwwwwwwwwwwwwwwwwww) [inline modules/haproxy/haproxy.tf]: TIL this is a thing! nice :)

**Comment by henrikhodne** [comment]
collectd-vsphere is proving to be kinda tricky since it depends on collectd, and Librato's collectd install script expires after a short while, so I need to emulate the steps of their install script.

File: modules/macstadium_go_worker/worker.conf.tpl [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+end script
+
+post-stop script
+  TRAVIS_WORKER_RUN_DIR=/var/tmp/run/travis-worker
+
+  if [ -f /etc/default/$UPSTART_JOB ]; then
+    . /etc/default/$UPSTART_JOB
+  fi
+
+  rm -f $TRAVIS_WORKER_RUN_DIR/$UPSTART_JOB-$INST
```
Comment (henrikhodne) [inline modules/macstadium_go_worker/worker.conf.tpl]: Accidentally copypastaed this from Jupiter Brain, `-$INST` should be removed

File: modules/vsphere_janitor/vsphere-janitor.conf.tpl [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+end script
+
+post-stop script
+  VSPHERE_JANITOR_RUN_DIR=/var/tmp/run/vsphere-janitor
+
+  if [ -f /etc/default/$UPSTART_JOB ]; then
+    . /etc/default/$UPSTART_JOB
+  fi
+
+  rm -f $VSPHERE_JANITOR_RUN_DIR/$UPSTART_JOB-$INST
```
Comment (henrikhodne) [inline modules/vsphere_janitor/vsphere-janitor.conf.tpl]: Same here, this `-$INST` also needs to be removed

File: modules/vsphere_janitor/worker.conf.tpl [hunk 4]
```diff
@@ -0,0 +1,38 @@
+description "Travis Worker (travis-worker-${env})"
```
Comment (henrikhodne) [inline modules/vsphere_janitor/worker.conf.tpl]: This entire file should probably be removed…

**Pull request merged**